### ISSUE TITLE
Future called twice with both

### ISSUE THREAD

**Issue by anthonyshull** [fluture-js/Fluture#118]
Using fluture 6.2.6

```js
import { both, node } from 'fluture';

const f1 = node((done) => {
  api({}, (err, msg) => done(err, msg));
});

const f2 = node((done) => {
  api({}, (err, msg) => done(err, msg));
});

both(f1, f2)
  .fork(err => console.error, res => console.log);
```

This was calling the api in f2 twice. Not sure if it was calling f1 more than once. Switched to:

```js
parallel(2, new Array(f1, f2))
  .fork(err => console.error, res => console.log);
```

And then it is working properly calling f1 and f2 once each.

**Comment by Avaq** [comment]
Thanks for reporting this. I was able to reproduce it in a unit test and will fix it immediately.

**Comment by Avaq** [comment]
The bug was introduced in f4341934edd9815f28d9b8ed5f85102c459f0f98

**Comment by Avaq** [comment]
@anthonyshull I've published a fix under `6.2.7`. If you get the chance, I'd be interested to hear whether this version resolves the issue within the context in which you first experienced it.
